tools/CMakeLists.txt

```diff
@@ -0,0 +1,51 @@
+# Create a header-only library from an SVD register definition file.
+#
+# add_groov_reg_lib(
+#     groov_reg_test 
+#     INPUT_FILE "rp2040.svd"
+#     OUTPUT_DIR ${CMAKE_CURRENT_BINARY_DIR}
+#     PERIPHERALS adc rtc
+# )
+
+find_package(Python3 COMPONENTS Interpreter)
+
+if(Python3_FOUND)
+    message(STATUS "Python3 Version: ${Python3_VERSION}")
+    message(STATUS "Python3 Executable: ${Python3_EXECUTABLE}")
+
+    function(add_groov_reg_lib target_name)
+        set(options "")
+        set(oneValueArgs INPUT_FILE OUTPUT_DIR)
+        set(multiValueArgs PERIPHERALS)
+        cmake_parse_arguments(PARSE_ARGV 1 ARG "${options}" "${oneValueArgs}" "${multiValueArgs}")
+
+        string(JOIN " " peripherals_string "${ARG_PERIPHERALS}")
+        set(python_script "${CMAKE_CURRENT_FUNCTION_LIST_DIR}/regs2groov.py")
+        set(register_headers)
+
+        set(register_headers)
+        foreach(name IN LISTS ARG_PERIPHERALS)
+            list(APPEND register_headers "${name}.hpp")
+        endforeach()
```

Comment:
**suggestion**:
`list(TRANSFORM ARG_PERIPHERALS APPEND ".hpp" OUTPUT_VARIABLE register_headers)`

Reply:
Note: The use of `cmake_parse_arguments(PARSE_ARGV, ...)` is necessitating the `IN LISTS` construction here I think? Simplifying that allows the list transform here.

Reply:
Looks like the suggestion works fine with argument parsing kept as is.